##### Copyright 2019 The TensorFlow Authors.

In [1]:
#@title Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

# Guia inicial de TensorFlow 2.0 para principiantes

<table class="tfo-notebook-buttons" align="left">
  <td>
    <a target="_blank" href="https://www.tensorflow.org/tutorials/quickstart/beginner"><img src="https://www.tensorflow.org/images/tf_logo_32px.png" />Ver en TensorFlow.org</a>
  </td>
  <td>
    <a target="_blank" href="https://colab.research.google.com/github/tensorflow/docs-l10n/blob/master/site/es-419/tutorials/quickstart/beginner.ipynb"><img src="https://www.tensorflow.org/images/colab_logo_32px.png" />Ejecutar en Google Colab</a>
  </td>
  <td>
    <a target="_blank" href="https://github.com/tensorflow/docs-l10n/blob/master/site/es-419/tutorials/quickstart/beginner.ipynb"><img src="https://www.tensorflow.org/images/GitHub-Mark-32px.png" />Ver codigo en GitHub</a>
  </td>
  <td>
    <a href="https://storage.googleapis.com/tensorflow_docs/docs-l10n/site/es-419/tutorials/quickstart/beginner.ipynb"><img src="https://www.tensorflow.org/images/download_logo_32px.png" />Descargar notebook</a>
  </td>
</table>

Note: Nuestra comunidad de Tensorflow ha traducido estos documentos. Como las traducciones de la comunidad
son basados en el "mejor esfuerzo", no hay ninguna garantia que esta sea un reflejo preciso y actual 
de la [Documentacion Oficial en Ingles](https://www.tensorflow.org/?hl=en).
Si tienen sugerencias sobre como mejorar esta traduccion, por favor envian un "Pull request"
al siguiente repositorio [tensorflow/docs](https://github.com/tensorflow/docs).
Para ofrecerse como voluntario o hacer revision de las traducciones de la Comunidad
por favor contacten al siguiente grupo [docs@tensorflow.org list](https://groups.google.com/a/tensorflow.org/forum/#!forum/docs).

Este es un notebook de [Google Colaboratory](https://colab.research.google.com/notebooks/welcome.ipynb). Los programas de Python se executan directamente en tu navegador— una gran manera de aprender y utilizar TensorFlow. 
Para poder seguir este tutorial, ejecuta este notebook en Google Colab presionando el boton en la parte superior de esta pagina.

1. En Colab, selecciona "connect to a Python runtime": En la parte superior derecha de la barra de menus selecciona: *CONNECT*.
2. Para ejecutar todas las celdas de este notebook:
 Selecciona *Runtime* > *Run all*.

Descarga e installa el paquete TensorFlow 2.0 version. Importa TensorFlow en tu programa:

In [2]:
# Installa TensorFlow

import tensorflow as tf

Carga y prepara el conjunto de datos [MNIST](http://yann.lecun.com/exdb/mnist/). Convierte los ejemplos de numeros enteros a numeros de punto flotante:

In [3]:
mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

Construye un modelo `tf.keras.Sequential`  apilando capas. Escoge un optimizador y una funcion de perdida para el entrenamiento de tu modelo:

In [4]:
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation='softmax')
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

Entrena y evalua el modelo:

In [5]:
model.fit(x_train, y_train, epochs=5)

model.evaluate(x_test,  y_test, verbose=2)

Epoch 1/5


   1/1875 [..............................] - ETA: 1s - loss: 2.4881 - accuracy: 0.0312

  31/1875 [..............................] - ETA: 3s - loss: 1.6096 - accuracy: 0.5484

  62/1875 [..............................] - ETA: 3s - loss: 1.2054 - accuracy: 0.6668

  93/1875 [>.............................] - ETA: 2s - loss: 1.0024 - accuracy: 0.7194

 123/1875 [>.............................] - ETA: 2s - loss: 0.8944 - accuracy: 0.7482

 153/1875 [=>............................] - ETA: 2s - loss: 0.8084 - accuracy: 0.7712

 182/1875 [=>............................] - ETA: 2s - loss: 0.7509 - accuracy: 0.7873

 213/1875 [==>...........................] - ETA: 2s - loss: 0.7043 - accuracy: 0.7999

 244/1875 [==>...........................] - ETA: 2s - loss: 0.6597 - accuracy: 0.8134

 275/1875 [===>..........................] - ETA: 2s - loss: 0.6263 - accuracy: 0.8233

 306/1875 [===>..........................] - ETA: 2s - loss: 0.5997 - accuracy: 0.8299

 336/1875 [====>.........................] - ETA: 2s - loss: 0.5748 - accuracy: 0.8371

 367/1875 [====>.........................] - ETA: 2s - loss: 0.5544 - accuracy: 0.8428

 398/1875 [=====>........................] - ETA: 2s - loss: 0.5363 - accuracy: 0.8475

 430/1875 [=====>........................] - ETA: 2s - loss: 0.5176 - accuracy: 0.8527

 460/1875 [======>.......................] - ETA: 2s - loss: 0.5022 - accuracy: 0.8562

 491/1875 [======>.......................] - ETA: 2s - loss: 0.4922 - accuracy: 0.8590

 522/1875 [=======>......................] - ETA: 2s - loss: 0.4826 - accuracy: 0.8621

 553/1875 [=======>......................] - ETA: 2s - loss: 0.4731 - accuracy: 0.8649

 584/1875 [========>.....................] - ETA: 2s - loss: 0.4633 - accuracy: 0.8676

 615/1875 [========>.....................] - ETA: 2s - loss: 0.4548 - accuracy: 0.8703

 647/1875 [=========>....................] - ETA: 2s - loss: 0.4474 - accuracy: 0.8721

 678/1875 [=========>....................] - ETA: 1s - loss: 0.4368 - accuracy: 0.8748

 709/1875 [==========>...................] - ETA: 1s - loss: 0.4296 - accuracy: 0.8769

 739/1875 [==========>...................] - ETA: 1s - loss: 0.4215 - accuracy: 0.8791

 772/1875 [===========>..................] - ETA: 1s - loss: 0.4148 - accuracy: 0.8812

 806/1875 [===========>..................] - ETA: 1s - loss: 0.4086 - accuracy: 0.8830

 839/1875 [============>.................] - ETA: 1s - loss: 0.4025 - accuracy: 0.8848

 872/1875 [============>.................] - ETA: 1s - loss: 0.3959 - accuracy: 0.8866

 905/1875 [=============>................] - ETA: 1s - loss: 0.3900 - accuracy: 0.8882

 938/1875 [==============>...............] - ETA: 1s - loss: 0.3857 - accuracy: 0.8895

 971/1875 [==============>...............] - ETA: 1s - loss: 0.3808 - accuracy: 0.8911

1004/1875 [===============>..............] - ETA: 1s - loss: 0.3753 - accuracy: 0.8927

1038/1875 [===============>..............] - ETA: 1s - loss: 0.3707 - accuracy: 0.8940

1071/1875 [================>.............] - ETA: 1s - loss: 0.3663 - accuracy: 0.8951

1105/1875 [================>.............] - ETA: 1s - loss: 0.3616 - accuracy: 0.8965

1139/1875 [=================>............] - ETA: 1s - loss: 0.3577 - accuracy: 0.8975

1173/1875 [=================>............] - ETA: 1s - loss: 0.3535 - accuracy: 0.8986

1206/1875 [==================>...........] - ETA: 1s - loss: 0.3494 - accuracy: 0.8996

1240/1875 [==================>...........] - ETA: 1s - loss: 0.3458 - accuracy: 0.9004

1274/1875 [===================>..........] - ETA: 0s - loss: 0.3430 - accuracy: 0.9009

1307/1875 [===================>..........] - ETA: 0s - loss: 0.3389 - accuracy: 0.9023

1341/1875 [====================>.........] - ETA: 0s - loss: 0.3353 - accuracy: 0.9032

1375/1875 [=====================>........] - ETA: 0s - loss: 0.3320 - accuracy: 0.9042

1409/1875 [=====================>........] - ETA: 0s - loss: 0.3295 - accuracy: 0.9051

1443/1875 [======================>.......] - ETA: 0s - loss: 0.3261 - accuracy: 0.9060

1477/1875 [======================>.......] - ETA: 0s - loss: 0.3227 - accuracy: 0.9071

1511/1875 [=======================>......] - ETA: 0s - loss: 0.3203 - accuracy: 0.9079

1545/1875 [=======================>......] - ETA: 0s - loss: 0.3174 - accuracy: 0.9089

1579/1875 [========================>.....] - ETA: 0s - loss: 0.3148 - accuracy: 0.9097

1612/1875 [========================>.....] - ETA: 0s - loss: 0.3115 - accuracy: 0.9106

1645/1875 [=========================>....] - ETA: 0s - loss: 0.3087 - accuracy: 0.9113

1678/1875 [=========================>....] - ETA: 0s - loss: 0.3064 - accuracy: 0.9119

1712/1875 [==========================>...] - ETA: 0s - loss: 0.3046 - accuracy: 0.9125

1746/1875 [==========================>...] - ETA: 0s - loss: 0.3023 - accuracy: 0.9132

1781/1875 [===========================>..] - ETA: 0s - loss: 0.2997 - accuracy: 0.9138

1815/1875 [============================>.] - ETA: 0s - loss: 0.2980 - accuracy: 0.9143

1849/1875 [============================>.] - ETA: 0s - loss: 0.2955 - accuracy: 0.9150

1875/1875 [==============================] - 3s 2ms/step - loss: 0.2935 - accuracy: 0.9155


Epoch 2/5
   1/1875 [..............................] - ETA: 0s - loss: 0.1981 - accuracy: 0.9375

  35/1875 [..............................] - ETA: 2s - loss: 0.1466 - accuracy: 0.9598

  69/1875 [>.............................] - ETA: 2s - loss: 0.1441 - accuracy: 0.9579

 104/1875 [>.............................] - ETA: 2s - loss: 0.1501 - accuracy: 0.9570

 139/1875 [=>............................] - ETA: 2s - loss: 0.1551 - accuracy: 0.9550

 173/1875 [=>............................] - ETA: 2s - loss: 0.1569 - accuracy: 0.9530

 207/1875 [==>...........................] - ETA: 2s - loss: 0.1545 - accuracy: 0.9540

 242/1875 [==>...........................] - ETA: 2s - loss: 0.1543 - accuracy: 0.9539

 276/1875 [===>..........................] - ETA: 2s - loss: 0.1513 - accuracy: 0.9546

 309/1875 [===>..........................] - ETA: 2s - loss: 0.1507 - accuracy: 0.9545

 343/1875 [====>.........................] - ETA: 2s - loss: 0.1505 - accuracy: 0.9547

 377/1875 [=====>........................] - ETA: 2s - loss: 0.1519 - accuracy: 0.9541

 410/1875 [=====>........................] - ETA: 2s - loss: 0.1515 - accuracy: 0.9542

 444/1875 [======>.......................] - ETA: 2s - loss: 0.1516 - accuracy: 0.9547

 477/1875 [======>.......................] - ETA: 2s - loss: 0.1507 - accuracy: 0.9547

 511/1875 [=======>......................] - ETA: 2s - loss: 0.1501 - accuracy: 0.9549

 545/1875 [=======>......................] - ETA: 1s - loss: 0.1503 - accuracy: 0.9550

 579/1875 [========>.....................] - ETA: 1s - loss: 0.1510 - accuracy: 0.9550

 613/1875 [========>.....................] - ETA: 1s - loss: 0.1516 - accuracy: 0.9552

 647/1875 [=========>....................] - ETA: 1s - loss: 0.1515 - accuracy: 0.9550

 681/1875 [=========>....................] - ETA: 1s - loss: 0.1524 - accuracy: 0.9544

 715/1875 [==========>...................] - ETA: 1s - loss: 0.1540 - accuracy: 0.9541

 749/1875 [==========>...................] - ETA: 1s - loss: 0.1544 - accuracy: 0.9539

 783/1875 [===========>..................] - ETA: 1s - loss: 0.1540 - accuracy: 0.9540

 816/1875 [============>.................] - ETA: 1s - loss: 0.1531 - accuracy: 0.9542

 850/1875 [============>.................] - ETA: 1s - loss: 0.1527 - accuracy: 0.9542

 884/1875 [=============>................] - ETA: 1s - loss: 0.1515 - accuracy: 0.9545

 918/1875 [=============>................] - ETA: 1s - loss: 0.1521 - accuracy: 0.9542

 952/1875 [==============>...............] - ETA: 1s - loss: 0.1515 - accuracy: 0.9545

 985/1875 [==============>...............] - ETA: 1s - loss: 0.1502 - accuracy: 0.9546

1019/1875 [===============>..............] - ETA: 1s - loss: 0.1505 - accuracy: 0.9545

1052/1875 [===============>..............] - ETA: 1s - loss: 0.1505 - accuracy: 0.9546

1085/1875 [================>.............] - ETA: 1s - loss: 0.1507 - accuracy: 0.9546

1118/1875 [================>.............] - ETA: 1s - loss: 0.1506 - accuracy: 0.9546

1151/1875 [=================>............] - ETA: 1s - loss: 0.1504 - accuracy: 0.9545

1185/1875 [=================>............] - ETA: 1s - loss: 0.1494 - accuracy: 0.9548

1219/1875 [==================>...........] - ETA: 0s - loss: 0.1487 - accuracy: 0.9549

1252/1875 [===================>..........] - ETA: 0s - loss: 0.1484 - accuracy: 0.9551

1287/1875 [===================>..........] - ETA: 0s - loss: 0.1478 - accuracy: 0.9550

1321/1875 [====================>.........] - ETA: 0s - loss: 0.1471 - accuracy: 0.9551

1355/1875 [====================>.........] - ETA: 0s - loss: 0.1471 - accuracy: 0.9552

1389/1875 [=====================>........] - ETA: 0s - loss: 0.1473 - accuracy: 0.9552

1423/1875 [=====================>........] - ETA: 0s - loss: 0.1471 - accuracy: 0.9553

1456/1875 [======================>.......] - ETA: 0s - loss: 0.1469 - accuracy: 0.9554

1490/1875 [======================>.......] - ETA: 0s - loss: 0.1462 - accuracy: 0.9556

1523/1875 [=======================>......] - ETA: 0s - loss: 0.1459 - accuracy: 0.9559

1557/1875 [=======================>......] - ETA: 0s - loss: 0.1461 - accuracy: 0.9561

1590/1875 [========================>.....] - ETA: 0s - loss: 0.1455 - accuracy: 0.9563

1623/1875 [========================>.....] - ETA: 0s - loss: 0.1448 - accuracy: 0.9566

1656/1875 [=========================>....] - ETA: 0s - loss: 0.1441 - accuracy: 0.9568

1689/1875 [==========================>...] - ETA: 0s - loss: 0.1439 - accuracy: 0.9569

1722/1875 [==========================>...] - ETA: 0s - loss: 0.1437 - accuracy: 0.9571

1755/1875 [===========================>..] - ETA: 0s - loss: 0.1435 - accuracy: 0.9571

1788/1875 [===========================>..] - ETA: 0s - loss: 0.1434 - accuracy: 0.9572

1821/1875 [============================>.] - ETA: 0s - loss: 0.1433 - accuracy: 0.9573

1855/1875 [============================>.] - ETA: 0s - loss: 0.1434 - accuracy: 0.9573

1875/1875 [==============================] - 3s 2ms/step - loss: 0.1430 - accuracy: 0.9573


Epoch 3/5
   1/1875 [..............................] - ETA: 0s - loss: 0.1963 - accuracy: 0.9062

  35/1875 [..............................] - ETA: 2s - loss: 0.1174 - accuracy: 0.9536

  69/1875 [>.............................] - ETA: 2s - loss: 0.1116 - accuracy: 0.9565

 103/1875 [>.............................] - ETA: 2s - loss: 0.1129 - accuracy: 0.9596

 137/1875 [=>............................] - ETA: 2s - loss: 0.1099 - accuracy: 0.9610

 171/1875 [=>............................] - ETA: 2s - loss: 0.1075 - accuracy: 0.9635

 205/1875 [==>...........................] - ETA: 2s - loss: 0.1135 - accuracy: 0.9627

 239/1875 [==>...........................] - ETA: 2s - loss: 0.1143 - accuracy: 0.9629

 273/1875 [===>..........................] - ETA: 2s - loss: 0.1147 - accuracy: 0.9625

 307/1875 [===>..........................] - ETA: 2s - loss: 0.1154 - accuracy: 0.9628

 341/1875 [====>.........................] - ETA: 2s - loss: 0.1155 - accuracy: 0.9630

 375/1875 [=====>........................] - ETA: 2s - loss: 0.1134 - accuracy: 0.9636

 409/1875 [=====>........................] - ETA: 2s - loss: 0.1122 - accuracy: 0.9641

 442/1875 [======>.......................] - ETA: 2s - loss: 0.1098 - accuracy: 0.9649

 476/1875 [======>.......................] - ETA: 2s - loss: 0.1117 - accuracy: 0.9644

 509/1875 [=======>......................] - ETA: 2s - loss: 0.1121 - accuracy: 0.9641

 542/1875 [=======>......................] - ETA: 2s - loss: 0.1106 - accuracy: 0.9651

 576/1875 [========>.....................] - ETA: 1s - loss: 0.1104 - accuracy: 0.9652

 610/1875 [========>.....................] - ETA: 1s - loss: 0.1109 - accuracy: 0.9650

 644/1875 [=========>....................] - ETA: 1s - loss: 0.1100 - accuracy: 0.9651

 678/1875 [=========>....................] - ETA: 1s - loss: 0.1099 - accuracy: 0.9654

 712/1875 [==========>...................] - ETA: 1s - loss: 0.1097 - accuracy: 0.9658

 746/1875 [==========>...................] - ETA: 1s - loss: 0.1098 - accuracy: 0.9659

 780/1875 [===========>..................] - ETA: 1s - loss: 0.1101 - accuracy: 0.9660

 813/1875 [============>.................] - ETA: 1s - loss: 0.1093 - accuracy: 0.9663

 847/1875 [============>.................] - ETA: 1s - loss: 0.1076 - accuracy: 0.9667

 880/1875 [=============>................] - ETA: 1s - loss: 0.1075 - accuracy: 0.9670

 914/1875 [=============>................] - ETA: 1s - loss: 0.1077 - accuracy: 0.9668

 947/1875 [==============>...............] - ETA: 1s - loss: 0.1072 - accuracy: 0.9667

 981/1875 [==============>...............] - ETA: 1s - loss: 0.1067 - accuracy: 0.9670

1015/1875 [===============>..............] - ETA: 1s - loss: 0.1064 - accuracy: 0.9672

1049/1875 [===============>..............] - ETA: 1s - loss: 0.1067 - accuracy: 0.9671

1083/1875 [================>.............] - ETA: 1s - loss: 0.1065 - accuracy: 0.9672

1117/1875 [================>.............] - ETA: 1s - loss: 0.1066 - accuracy: 0.9673

1151/1875 [=================>............] - ETA: 1s - loss: 0.1060 - accuracy: 0.9674

1185/1875 [=================>............] - ETA: 1s - loss: 0.1059 - accuracy: 0.9674

1218/1875 [==================>...........] - ETA: 0s - loss: 0.1061 - accuracy: 0.9674

1251/1875 [===================>..........] - ETA: 0s - loss: 0.1064 - accuracy: 0.9674

1285/1875 [===================>..........] - ETA: 0s - loss: 0.1070 - accuracy: 0.9670

1319/1875 [====================>.........] - ETA: 0s - loss: 0.1069 - accuracy: 0.9671

1353/1875 [====================>.........] - ETA: 0s - loss: 0.1072 - accuracy: 0.9668

1387/1875 [=====================>........] - ETA: 0s - loss: 0.1072 - accuracy: 0.9668

1421/1875 [=====================>........] - ETA: 0s - loss: 0.1076 - accuracy: 0.9668

1455/1875 [======================>.......] - ETA: 0s - loss: 0.1075 - accuracy: 0.9669

1489/1875 [======================>.......] - ETA: 0s - loss: 0.1069 - accuracy: 0.9670

1522/1875 [=======================>......] - ETA: 0s - loss: 0.1074 - accuracy: 0.9668

1556/1875 [=======================>......] - ETA: 0s - loss: 0.1074 - accuracy: 0.9668

1589/1875 [========================>.....] - ETA: 0s - loss: 0.1076 - accuracy: 0.9667

1622/1875 [========================>.....] - ETA: 0s - loss: 0.1077 - accuracy: 0.9668

1655/1875 [=========================>....] - ETA: 0s - loss: 0.1079 - accuracy: 0.9668

1689/1875 [==========================>...] - ETA: 0s - loss: 0.1076 - accuracy: 0.9669

1723/1875 [==========================>...] - ETA: 0s - loss: 0.1070 - accuracy: 0.9671

1757/1875 [===========================>..] - ETA: 0s - loss: 0.1070 - accuracy: 0.9672

1791/1875 [===========================>..] - ETA: 0s - loss: 0.1066 - accuracy: 0.9672

1825/1875 [============================>.] - ETA: 0s - loss: 0.1075 - accuracy: 0.9671

1858/1875 [============================>.] - ETA: 0s - loss: 0.1075 - accuracy: 0.9670

1875/1875 [==============================] - 3s 2ms/step - loss: 0.1073 - accuracy: 0.9670


Epoch 4/5
   1/1875 [..............................] - ETA: 0s - loss: 0.0097 - accuracy: 1.0000

  35/1875 [..............................] - ETA: 2s - loss: 0.0646 - accuracy: 0.9759

  69/1875 [>.............................] - ETA: 2s - loss: 0.0763 - accuracy: 0.9737

 103/1875 [>.............................] - ETA: 2s - loss: 0.0826 - accuracy: 0.9712

 137/1875 [=>............................] - ETA: 2s - loss: 0.0878 - accuracy: 0.9719

 171/1875 [=>............................] - ETA: 2s - loss: 0.0832 - accuracy: 0.9740

 206/1875 [==>...........................] - ETA: 2s - loss: 0.0848 - accuracy: 0.9742

 240/1875 [==>...........................] - ETA: 2s - loss: 0.0867 - accuracy: 0.9728

 273/1875 [===>..........................] - ETA: 2s - loss: 0.0894 - accuracy: 0.9724

 307/1875 [===>..........................] - ETA: 2s - loss: 0.0907 - accuracy: 0.9722

 341/1875 [====>.........................] - ETA: 2s - loss: 0.0897 - accuracy: 0.9724

 375/1875 [=====>........................] - ETA: 2s - loss: 0.0917 - accuracy: 0.9719

 408/1875 [=====>........................] - ETA: 2s - loss: 0.0899 - accuracy: 0.9720

 442/1875 [======>.......................] - ETA: 2s - loss: 0.0890 - accuracy: 0.9725

 475/1875 [======>.......................] - ETA: 2s - loss: 0.0878 - accuracy: 0.9730

 509/1875 [=======>......................] - ETA: 2s - loss: 0.0882 - accuracy: 0.9729

 542/1875 [=======>......................] - ETA: 1s - loss: 0.0875 - accuracy: 0.9731

 576/1875 [========>.....................] - ETA: 1s - loss: 0.0872 - accuracy: 0.9732

 609/1875 [========>.....................] - ETA: 1s - loss: 0.0875 - accuracy: 0.9728

 642/1875 [=========>....................] - ETA: 1s - loss: 0.0872 - accuracy: 0.9726

 676/1875 [=========>....................] - ETA: 1s - loss: 0.0876 - accuracy: 0.9725

 710/1875 [==========>...................] - ETA: 1s - loss: 0.0867 - accuracy: 0.9727

 744/1875 [==========>...................] - ETA: 1s - loss: 0.0859 - accuracy: 0.9730

 778/1875 [===========>..................] - ETA: 1s - loss: 0.0858 - accuracy: 0.9730

 812/1875 [===========>..................] - ETA: 1s - loss: 0.0872 - accuracy: 0.9727

 846/1875 [============>.................] - ETA: 1s - loss: 0.0872 - accuracy: 0.9727

 880/1875 [=============>................] - ETA: 1s - loss: 0.0877 - accuracy: 0.9725

 914/1875 [=============>................] - ETA: 1s - loss: 0.0876 - accuracy: 0.9725

 948/1875 [==============>...............] - ETA: 1s - loss: 0.0876 - accuracy: 0.9725

 982/1875 [==============>...............] - ETA: 1s - loss: 0.0873 - accuracy: 0.9725

1016/1875 [===============>..............] - ETA: 1s - loss: 0.0877 - accuracy: 0.9726

1049/1875 [===============>..............] - ETA: 1s - loss: 0.0875 - accuracy: 0.9725

1083/1875 [================>.............] - ETA: 1s - loss: 0.0880 - accuracy: 0.9722

1117/1875 [================>.............] - ETA: 1s - loss: 0.0879 - accuracy: 0.9724

1151/1875 [=================>............] - ETA: 1s - loss: 0.0884 - accuracy: 0.9722

1185/1875 [=================>............] - ETA: 1s - loss: 0.0887 - accuracy: 0.9721

1218/1875 [==================>...........] - ETA: 0s - loss: 0.0887 - accuracy: 0.9721

1252/1875 [===================>..........] - ETA: 0s - loss: 0.0884 - accuracy: 0.9723

1285/1875 [===================>..........] - ETA: 0s - loss: 0.0888 - accuracy: 0.9721

1318/1875 [====================>.........] - ETA: 0s - loss: 0.0890 - accuracy: 0.9720

1351/1875 [====================>.........] - ETA: 0s - loss: 0.0883 - accuracy: 0.9722

1385/1875 [=====================>........] - ETA: 0s - loss: 0.0883 - accuracy: 0.9722

1419/1875 [=====================>........] - ETA: 0s - loss: 0.0882 - accuracy: 0.9723

1452/1875 [======================>.......] - ETA: 0s - loss: 0.0883 - accuracy: 0.9722

1486/1875 [======================>.......] - ETA: 0s - loss: 0.0881 - accuracy: 0.9723

1520/1875 [=======================>......] - ETA: 0s - loss: 0.0878 - accuracy: 0.9724

1554/1875 [=======================>......] - ETA: 0s - loss: 0.0883 - accuracy: 0.9722

1588/1875 [========================>.....] - ETA: 0s - loss: 0.0890 - accuracy: 0.9720

1621/1875 [========================>.....] - ETA: 0s - loss: 0.0888 - accuracy: 0.9722

1655/1875 [=========================>....] - ETA: 0s - loss: 0.0884 - accuracy: 0.9723

1688/1875 [==========================>...] - ETA: 0s - loss: 0.0885 - accuracy: 0.9722

1722/1875 [==========================>...] - ETA: 0s - loss: 0.0882 - accuracy: 0.9724

1755/1875 [===========================>..] - ETA: 0s - loss: 0.0878 - accuracy: 0.9725

1789/1875 [===========================>..] - ETA: 0s - loss: 0.0882 - accuracy: 0.9725

1822/1875 [============================>.] - ETA: 0s - loss: 0.0881 - accuracy: 0.9725

1856/1875 [============================>.] - ETA: 0s - loss: 0.0881 - accuracy: 0.9724

1875/1875 [==============================] - 3s 2ms/step - loss: 0.0880 - accuracy: 0.9725


Epoch 5/5
   1/1875 [..............................] - ETA: 0s - loss: 0.0140 - accuracy: 1.0000

  34/1875 [..............................] - ETA: 2s - loss: 0.0573 - accuracy: 0.9789

  67/1875 [>.............................] - ETA: 2s - loss: 0.0667 - accuracy: 0.9799

 100/1875 [>.............................] - ETA: 2s - loss: 0.0635 - accuracy: 0.9812

 133/1875 [=>............................] - ETA: 2s - loss: 0.0662 - accuracy: 0.9805

 166/1875 [=>............................] - ETA: 2s - loss: 0.0661 - accuracy: 0.9804

 199/1875 [==>...........................] - ETA: 2s - loss: 0.0645 - accuracy: 0.9804

 232/1875 [==>...........................] - ETA: 2s - loss: 0.0647 - accuracy: 0.9806

 265/1875 [===>..........................] - ETA: 2s - loss: 0.0662 - accuracy: 0.9800

 298/1875 [===>..........................] - ETA: 2s - loss: 0.0681 - accuracy: 0.9794

 330/1875 [====>.........................] - ETA: 2s - loss: 0.0686 - accuracy: 0.9791

 363/1875 [====>.........................] - ETA: 2s - loss: 0.0691 - accuracy: 0.9787

 396/1875 [=====>........................] - ETA: 2s - loss: 0.0682 - accuracy: 0.9790

 430/1875 [=====>........................] - ETA: 2s - loss: 0.0686 - accuracy: 0.9789

 463/1875 [======>.......................] - ETA: 2s - loss: 0.0677 - accuracy: 0.9789

 496/1875 [======>.......................] - ETA: 2s - loss: 0.0683 - accuracy: 0.9785

 529/1875 [=======>......................] - ETA: 2s - loss: 0.0687 - accuracy: 0.9785

 562/1875 [=======>......................] - ETA: 2s - loss: 0.0687 - accuracy: 0.9783

 595/1875 [========>.....................] - ETA: 1s - loss: 0.0695 - accuracy: 0.9779

 628/1875 [=========>....................] - ETA: 1s - loss: 0.0708 - accuracy: 0.9776

 661/1875 [=========>....................] - ETA: 1s - loss: 0.0706 - accuracy: 0.9777

 694/1875 [==========>...................] - ETA: 1s - loss: 0.0704 - accuracy: 0.9778

 727/1875 [==========>...................] - ETA: 1s - loss: 0.0704 - accuracy: 0.9776

 760/1875 [===========>..................] - ETA: 1s - loss: 0.0701 - accuracy: 0.9776

 793/1875 [===========>..................] - ETA: 1s - loss: 0.0704 - accuracy: 0.9774

 827/1875 [============>.................] - ETA: 1s - loss: 0.0711 - accuracy: 0.9771

 860/1875 [============>.................] - ETA: 1s - loss: 0.0720 - accuracy: 0.9768

 894/1875 [=============>................] - ETA: 1s - loss: 0.0720 - accuracy: 0.9768

 928/1875 [=============>................] - ETA: 1s - loss: 0.0723 - accuracy: 0.9768

 961/1875 [==============>...............] - ETA: 1s - loss: 0.0727 - accuracy: 0.9766

 994/1875 [==============>...............] - ETA: 1s - loss: 0.0731 - accuracy: 0.9763

1028/1875 [===============>..............] - ETA: 1s - loss: 0.0737 - accuracy: 0.9760

1061/1875 [===============>..............] - ETA: 1s - loss: 0.0735 - accuracy: 0.9762

1094/1875 [================>.............] - ETA: 1s - loss: 0.0744 - accuracy: 0.9761

1128/1875 [=================>............] - ETA: 1s - loss: 0.0742 - accuracy: 0.9761

1161/1875 [=================>............] - ETA: 1s - loss: 0.0742 - accuracy: 0.9761

1194/1875 [==================>...........] - ETA: 1s - loss: 0.0744 - accuracy: 0.9760

1227/1875 [==================>...........] - ETA: 0s - loss: 0.0742 - accuracy: 0.9760

1260/1875 [===================>..........] - ETA: 0s - loss: 0.0746 - accuracy: 0.9758

1293/1875 [===================>..........] - ETA: 0s - loss: 0.0750 - accuracy: 0.9757

1326/1875 [====================>.........] - ETA: 0s - loss: 0.0754 - accuracy: 0.9756

1359/1875 [====================>.........] - ETA: 0s - loss: 0.0754 - accuracy: 0.9756

1392/1875 [=====================>........] - ETA: 0s - loss: 0.0747 - accuracy: 0.9758

1426/1875 [=====================>........] - ETA: 0s - loss: 0.0754 - accuracy: 0.9758

1459/1875 [======================>.......] - ETA: 0s - loss: 0.0753 - accuracy: 0.9759

1491/1875 [======================>.......] - ETA: 0s - loss: 0.0753 - accuracy: 0.9759

1524/1875 [=======================>......] - ETA: 0s - loss: 0.0755 - accuracy: 0.9759

1557/1875 [=======================>......] - ETA: 0s - loss: 0.0750 - accuracy: 0.9759

1590/1875 [========================>.....] - ETA: 0s - loss: 0.0749 - accuracy: 0.9760

1623/1875 [========================>.....] - ETA: 0s - loss: 0.0748 - accuracy: 0.9761

1656/1875 [=========================>....] - ETA: 0s - loss: 0.0747 - accuracy: 0.9761

1689/1875 [==========================>...] - ETA: 0s - loss: 0.0748 - accuracy: 0.9761

1722/1875 [==========================>...] - ETA: 0s - loss: 0.0747 - accuracy: 0.9760

1755/1875 [===========================>..] - ETA: 0s - loss: 0.0745 - accuracy: 0.9761

1789/1875 [===========================>..] - ETA: 0s - loss: 0.0749 - accuracy: 0.9760

1822/1875 [============================>.] - ETA: 0s - loss: 0.0747 - accuracy: 0.9760

1856/1875 [============================>.] - ETA: 0s - loss: 0.0745 - accuracy: 0.9761

1875/1875 [==============================] - 3s 2ms/step - loss: 0.0745 - accuracy: 0.9761


313/313 - 0s - loss: 0.0707 - accuracy: 0.9785


[0.07073566317558289, 0.9785000085830688]

El model de clasificacion de images fue entrenado y alcanzo una exactitud de ~98% en este conjunto de datos. Para aprender mas, lee los [tutoriales de TensorFlow].